src/components/KeywordHighlighterComponent.js

```diff
@@ -0,0 +1,255 @@
+import {
+  EventHandlerService,
+  PDFLEvents,
+} from "../services/EventHandlerService";
+
+/**
+ * Used for marking the text that can be highlighted.
+ */
+const TOPIC_HIGHLIGHTED_TEXT_CLASS = "topic-highlighted-text";
+/**
+ * Used for turning the highlight on/off.
+ */
+const TOPIC_HIGHLIGHT_ACTIVE_CLASS = "topic-highlighted";
+
+/**
+ * Listens for 'onTextLayerRendered' events and highlights the text using some keywords.
+ * Sentences that should be highlighted are marked by 'span' html elements with class
+ * 'topic-highlighted-text'.
+ *
+ * Class 'topic-highlighted' actually contains css that highlights them.
+ * 2 classes are used so highlighting can be turned off.
+ *
+ * @property {String[]} keywords a list of keywords that are going to be used for
+ *  highlighting
+ * @property {bool} on flag which is going to be used to check whether to highlight
+ *  the appropriate sentences
+ */
+class KeywordHighlighterComponent {
+  constructor() {
+    this.keywords = [];
+    this.on = false;
+
+    this.#registerEvents();
+  }
+
+  /**
+   * Sets the keywords that the highlighting is going to use.
+   *
+   * @param {String[]} keywords
+   */
+  setKeywords(keywords) {
+    this.keywords = keywords;
+  }
+
+  /**
+   * Shows the highlited sentences, text layers processed next will have their
+   * highlited sentences turned on by default.
+   */
+  turnOn() {
+    this.on = true;
```

Comment:
It is better to set the state only after the computation ends.
Nothing important obviously since it works in the same way at the end

Reply:
Done